### ISSUE TITLE
Neo blocks randomly deleted on all multisite sites

### ISSUE THREAD

**Issue by lenvanessen** [spicywebau/craft-neo#631]
#### Description
We upgraded to 3.2.3 hoping it would solve some of these weird errors. But unfortunately not, so we're creating a ticket to describe some of the behavior we're experiencing.

We have a pageparts field that that has the following config:
```
columnSuffix: null
contentColumnType: string
fieldGroup: 0d454b1a-5e3c-46f6-a4c0-bb76d1a049ca # Common
handle: pageParts
instructions: null
name: 'Page Parts'
searchable: false
settings:
  maxBlocks: null
  maxLevels: null
  maxTopBlocks: null
  minBlocks: null
  propagationKeyFormat: null
  propagationMethod: all
translationKeyFormat: null
translationMethod: site
type: benf\neo\Field
```

We started our site with 2 sites, and everything worked fine.  But as soon as we added 2 additional sites, everything went haywire. The blocks are correctly synced between the 2 original sites, but as soon as you start duplicating pages that had existed on the original site, to one of the new sites, problems start. When you open the newly propagated page, it misses most of the blocks. When you then save that entry for that site, the other blocks on other sites are deleted.

Our section is set to "Let each entry decide what site to propagate to"

#### Steps to reproduce
1. Create sites in Craft 4.1.2
2. Create a section
3. Create some neo blocks
4. Save an entry with some blocks
5. Add a new site to Craft
6. Propagate the page to that new site
7. Check the blocks

#### Other information
- Neo version: 3.2.3
- Craft version: 4.1.2
- (if multi-site) What is the affected Neo field's propagation method: all sitess
- Is eager-loading used? no

We'd be more than happy to provide a db example and composer.json

**Comment by ttempleton** [comment]
Could you please send that database backup and composer.json to plugins@spicyweb.com.au and we'll have a look.

**Comment by lenvanessen** [comment]
@ttempleton you got mail!

**Comment by rene-haefliger** [comment]
The error is now no longer present. Thanks!

**Comment by lenvanessen** [comment]
@rene-haefliger what error? I haven't seen a commit that addresses this issue?

**Comment by rene-haefliger** [comment]
Sorry, wrong issue.

**Comment by lenvanessen** [comment]
@rene-haefliger you got me exited there for a second, I saw your message in my inbox and though it was fixed:)

**Comment by lenvanessen** [comment]
@ttempleton  any update or in-between fix we can apply? We currently cannot update the website without neo-fields disappearing

**Comment by ttempleton** [comment]
Thanks for sending that through @lenvanessen. I reproduced the block loss once, but not reliably - when it happened, I saw that the missing blocks were missing because they didn't have rows added to the `elements_sites` table. Neo doesn't do anything to touch the `elements_sites` table directly, so I'm not sure if this is actually being caused by a Neo bug, but I'm guessing since I couldn't reliably reproduce it, it could be that the entry is being manually saved too soon before Craft itself has finished resaving the entries for the new site.

I've just released Neo 3.2.4 which adds the `--with-propagation-method` option to the `php craft neo/fields/reapply-propagation-method` console command. When I experienced the block loss, running...

```
php craft neo/fields/reapply-propagation-method --with-propagation-method=all
```

...fixed the Neo field on the new site. Please do try that command and let me know if it works for you.

**Comment by lenvanessen** [comment]
@ttempleton the process of Craft saving the entries to a new site would be in the queue right? Our queue generally is quite busy, so it could definitely be the case. I'm not sure that a busy queue is unique for our project, or if the entry should be somehow locked in the other languages until Craft has finished propagating (or done immediately, the author would need to wait a bit longer, but you'd have more consistency across sites).

But I guess this is more a discussion for Craft, rather than for Neo. In any case, it would be the `PropagateElements` job right?

For our purpose, I guess we then could manually execute it immediately like so, so we're it was done propagating:
```
  Event::on(
            Queue::class,
            Queue::EVENT_BEFORE_PUSH,
            function(PushEvent $event) {
                if($event->job instanceof PropagateElements) {
                    $job->execute(Craft::$app->getQueue());
                    $event->handled =  true;
                }
            }
        );

```

**Comment by ttempleton** [comment]
Hmm, I just retested the command to make sure, and it's definitely working for me. Try to remove `vendor/` and reinstall your Composer dependencies and let me know if that works.

The command uses the `ApplyNewPropagationMethod` job.

**Comment by lenvanessen** [comment]
@ttempleton sorry, my bad, in any case, I successfully ran the command, but still missing blocks.

What I did was:
1. Run the command and let it execute in the queue

Then in the CMS
1. Delete the "Over ons" entry on the Spanish site
2. Re-add it trough the dutch site, by using "add to site" under status on the right-hand site
3. Wait for the queue
4. Check entry on spanish site, most blocks where missing
5. Run your command, it said the output 

I'll send you a screenshare per mail so you can see, for us it's consistant, 100% of the times we follow the steps above, blocks are missing

**Comment by ttempleton** [comment]
Thanks, with those steps I was able to reliably reproduce it, then with what I learned there I was able to reproduce it on my usual testing environment. It turned out that the only block(s) that still appeared on the newly enabled site were those with a link field, and that was happening because the link fields were being incorrectly detected as being modified. I also tested this with a Matrix field - one block type with a link field, one block type without - and the same thing happened with the link fields being detected as modified, and the blocks without a link field not appearing on the new site. I'm not certain if that's a bug in Craft or the Typed Link Field plugin (though I'm definitely leaning towards Typed Link Field) but at this point it doesn't seem like a Neo bug.
